### PR TITLE
Deprecate api_token in favor of api_key

### PR DESCRIPTION
**Pull request by caliban0**
See https://github.com/cherryservers/cherryctl/issues/73 for context.

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -46,9 +52,24 @@ func (p *CherryServersProvider) Schema(ctx context.Context, req provider.SchemaR
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"api_token": schema.StringAttribute{
-				Description: "Cherry Servers [API Key](https://portal.cherryservers.com/settings/api-keys) that allows interactions with the API.",
-				Optional:    true,
-				Sensitive:   true,
+				Description: "**Deprecated**: use `api_key` instead, as this attribute is deprecated " +
+					"and will removed in the next major version of the provider. " +
```
Comment (zalmarge) [inline internal/provider/provider.go]: Should be "will **be** removed".

File: internal/provider/provider.go [hunk 2]
```diff
@@ -46,9 +52,24 @@ func (p *CherryServersProvider) Schema(ctx context.Context, req provider.SchemaR
...
+					"Cherry Servers [API Key](https://portal.cherryservers.com/settings/api-keys)" +
+					"that allows interactions with the API.",
+				Optional:  true,
+				Sensitive: true,
+				DeprecationMessage: "Use `api_key` instead, as this attribute is deprecated " +
+					"and will removed in the next major version of the provider.",
```
Comment (zalmarge) [inline internal/provider/provider.go]: Should be "will **be** removed".

File: internal/provider/provider.go [hunk 3]
```diff
@@ -46,9 +52,24 @@ func (p *CherryServersProvider) Schema(ctx context.Context, req provider.SchemaR
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"api_token": schema.StringAttribute{
-				Description: "Cherry Servers [API Key](https://portal.cherryservers.com/settings/api-keys) that allows interactions with the API.",
-				Optional:    true,
-				Sensitive:   true,
+				Description: "**Deprecated**: use `api_key` instead, as this attribute is deprecated " +
+					"and will removed in the next major version of the provider. " +
+					"Cherry Servers [API Key](https://portal.cherryservers.com/settings/api-keys)" +
```
Comment (zalmarge) [inline internal/provider/provider.go]: Missing space in concatenated description: "...api-keys)" + "that allows..." renders as "...api-keys)that allows..."

File: internal/provider/provider.go [hunk 4]
```diff
@@ -46,9 +52,24 @@ func (p *CherryServersProvider) Schema(ctx context.Context, req provider.SchemaR
...
+					"that allows interactions with the API.",
+				Optional:  true,
+				Sensitive: true,
+				DeprecationMessage: "Use `api_key` instead, as this attribute is deprecated " +
+					"and will removed in the next major version of the provider.",
+			},
+			"api_key": schema.StringAttribute{
+				Description: "Cherry Servers [API Key](https://portal.cherryservers.com/settings/api-keys)" +
```
Comment (zalmarge) [inline internal/provider/provider.go]: Missing space in concatenated description: "...api-keys)" + fmt.Sprintf(...

File: internal/provider/provider.go [hunk 5]
```diff
@@ -65,53 +86,85 @@ func (p *CherryServersProvider) Configure(ctx context.Context, req provider.Conf
...
-			"Missing CherryServers API Token",
-			"The provider cannot create the CherryServers API client as there is a missing or empty value for the CherryServers API token. "+
-				"Set the API token value in the configuration or use the CHERRY_AUTH_TOKEN or CHERRY_AUTH_KEY environment variables. "+
-				"If either is already set, ensure the value is not empty.",
+			path.Root("api_key"),
+			"Missing CherryServers API key",
+			"The provider cannot create the CherryServers API client "+
+				"as there is a missing or empty value for the CherryServers API key. "+
+				"Set the API key value in the configuration or use the "+
+				fmt.Sprintf("%s environment variables. ", apiKeyVar),
```
Comment (zalmarge) [inline internal/provider/provider.go]: Shouldn't a singular form be used here? We have only one environment variable to set API key value now, no?

File: internal/provider/provider.go [hunk 6]
```diff
@@ -65,53 +86,85 @@ func (p *CherryServersProvider) Configure(ctx context.Context, req provider.Conf
...
+			fmt.Sprintf("%s is deprecated and will be removed in the next major ", source)+
+				fmt.Sprintf("version of the provider, please use %s instead.", apiKeyVar))
+	}
+
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	ctx = tflog.SetField(ctx, "cherryservers_api_token", apiToken)
+	ctx = tflog.SetField(ctx, "cherryservers_api_token", apiKey)
```
Comment (zalmarge) [inline internal/provider/provider.go]: Still using a field key name with a "token" in it. Not sure if that's by intention or not.

**Review by zalmarge — CHANGES_REQUESTED** [review]
Mostly minor issues, see comments below.

Also, no test exercises the new Configure precedence logic — none of the six paths (env CHERRY_AUTH_KEY, CHERRY_AUTH_TOKEN, CHERRY_API_KEY, config api_token, config api_key, both-set conflict, deprecation warning emitted) has a unit test. 

Acceptance test setup (sweeper_test.go) and CI (test.yml) were updated to use CHERRY_API_KEY, but this only proves the new var works, not that the deprecated paths and precedence still behave correctly and does not prevent from silent regressions.

**Comment by caliban0** [comment]
Once https://github.com/cherryservers/terraform-provider-cherryservers/pull/114 is merged and `v1.6.0` is synced with `master`, I'll push the corrections for the index template.

File: internal/provider/provider.go (hunk 1 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

File: internal/provider/provider.go (hunk 2 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

File: internal/provider/provider.go (hunk 3 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

File: internal/provider/provider.go (hunk 4 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

File: internal/provider/provider.go (hunk 5 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

File: internal/provider/provider.go (hunk 6 above)
Comment (caliban0) [reply internal/provider/provider.go]: Fixed.

**Comment by caliban0** [comment]
Added unit tests, along with some refactoring to facilitate them.

File: internal/provider/sweeper_test.go [hunk 7]
```diff
@@ -19,21 +20,18 @@ func TestMain(m *testing.M) {
 
 // sharedClient returns a common provider client.
 func sharedClient() (any, error) {
-	apiKey := os.Getenv("CHERRY_AUTH_KEY")
-	if apiKey == "" {
-		apiKey = os.Getenv("CHERRY_AUTH_TOKEN")
-	}
+	apiKey := os.Getenv(apiKeyVar)
```
Comment (zalmarge) [inline internal/provider/sweeper_test.go:25]: Is it really OK to remove the fallback to the old environment variable? After all it's just deprecated, not removed.

File: internal/provider/provider.go [hunk 8]
```diff
@@ -46,72 +57,130 @@ func (p *CherryServersProvider) Schema(ctx context.Context, req provider.SchemaR
...
+	if diags.HasError() {
+		return ""
 	}
 
-	apiToken := os.Getenv("CHERRY_AUTH_KEY")
-	if apiToken == "" {
-		apiToken = os.Getenv("CHERRY_AUTH_TOKEN")
+	var source, key string
+
+	// CHERRY_AUTH_TOKEN and CHERRY_AUTH_KEY are deprecated,
```
Comment (zalmarge) [inline internal/provider/provider.go:112]: Just a suggestion. When I have more than two obsoleted settings/envvars, I usually run a loop over the list of them instead. That also lets adding another depricated setting/envvar easily. Something like that:

```go
// Listed lowest to highest precedence: CHERRY_AUTH_TOKEN and
// CHERRY_AUTH_KEY are deprecated, so CHERRY_API_KEY beats them.
for _, envVar := range []string{oldestApiKeyVar, oldApiKeyVar, apiKeyVar} {
	if k := os.Getenv(envVar); k != "" {
		key = k
		source = envVar
	}
}
```

Haven't tried to compile, but I think you get the idea.

Comment (caliban0) [reply internal/provider/provider.go:112]: Yeah, that's definitely cleaner. Refactored.

File: internal/provider/sweeper_test.go (hunk 7 above)
Comment (caliban0) [reply internal/provider/sweeper_test.go:25]: I think it's fine. This is a test, not user user facing code, so it only affects developers.

**Review by zalmarge — APPROVED** [review]
